spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1594,6 +1598,34 @@ public void onMessage(Message message) {
 		}
 	}
 
+	private QueueingConsumer createQueueingConsumer(final String queueName, Channel channel) throws Exception {
+		channel.basicQos(1);
+		final CountDownLatch latch = new CountDownLatch(1);
+		QueueingConsumer consumer = new QueueingConsumer(channel) {
+
+			@Override
+			public void handleCancel(String consumerTag) throws IOException {
+				super.handleCancel(consumerTag);
+				latch.countDown();
```

Comment:
Good. Sorry, if I wasn't clear here.
I just mean that we don't need this `handleCancel` and interaction on the `latch`.
I agreed about the `channel.close()` after some reasonable timeout.

From other side this `handleCancel()` doesn't hurt and we can still live with it because of GA dead line :smile: .
So, merging after squash and rebase!